go/inst/cluster_alias_dao.go

```diff
@@ -114,65 +114,102 @@ func writeClusterAliasManualOverride(clusterName string, alias string) error {
 	return ExecDBWriteFunc(writeFunc)
 }
 
+// Original, safe approach, which uses REPLACE INTO
+func updateClusterAliasesUsingReplace() error {
+	_, err := db.ExecOrchestrator(`
+	replace into
+			cluster_alias (alias, cluster_name, last_registered)
+		select
+			suggested_cluster_alias,
+				cluster_name,
+				now()
+			from
+			database_instance
+			left join database_instance_downtime using (hostname, port)
+		  where
+			suggested_cluster_alias!=''
+				/* exclude newly demoted, downtimed masters */
+				and ifnull(
+						database_instance_downtime.downtime_active = 1
+						and database_instance_downtime.end_timestamp > now()
+						and database_instance_downtime.reason = ?
+					, 0) = 0
+			order by
+				ifnull(last_checked <= last_seen, 0) asc,
+				read_only desc,
+				num_slave_hosts asc
+	`, DowntimeLostInRecoveryMessage)
+
+	return err
+}
+
+// Optimized approach using INSERT INTO ... ON DUPLICATE KEY UPDATE
+// While this approach is much faster and works in most cases, it is not
+// guaranteed to be working in every case.
+// cluster_alias table has two unique indexes:
+// 1. primary on cluster_name column
+// 2. alias_uidx on alias colum
+//
+// The data which is going to be inserted originates from database_instance
+// table, in particular the following columns:
+// 1. `cluster_name` varchar(128) NOT NULL
+// 2. `suggested_cluster_alias` varchar(128) CHARACTER SET ascii COLLATE
```

Comment:
is there a unique index on suggested_cluster_alias on database_instance table? we discussed the very same issue and we thought there cannot be duplicate alias name?

Or may be it was the cluster_name.

Reply:
No unique index on `suggested_cluster_alias`

Reply:
Thanks for the detailed analysis on JIRA. Agree, we can allow to have conflicting aliases but it is questionable behavior and confusing to users.